drivers/sensor/veml7700/veml7700.c

```diff
@@ -535,30 +499,70 @@ static int veml7700_init(const struct device *dev)
 	return 0;
 }
 
+#ifdef CONFIG_PM_DEVICE
+static int veml7700_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	const struct veml7700_config *conf = dev->config;
+	struct veml7700_data *data = dev->data;
+	uint8_t prev_sd;
+	int ret;
+
+	if (conf->psm != VEML7700_PSM_DISABLED) {
+		switch (action) {
+		case PM_DEVICE_ACTION_SUSPEND:
+			return veml7700_set_shutdown_flag(dev, 1);
+
+		case PM_DEVICE_ACTION_RESUME:
+			return veml7700_set_shutdown_flag(dev, 0);
+
+		case PM_DEVICE_ACTION_TURN_OFF:
+			return 0;
+
+		case PM_DEVICE_ACTION_TURN_ON:
+			prev_sd = data->shut_down;
+			data->shut_down = 0;
+			ret = veml7700_init(dev);
+			if (ret < 0) {
+				data->shut_down = prev_sd;
+				return ret;
+			}
+			return veml7700_set_shutdown_flag(dev, prev_sd);
+
+		default:
+			return -ENOTSUP;
+		}
+	}
+
+	return 0;
+}
+#endif /* CONFIG_PM_DEVICE */
+
 static const struct sensor_driver_api veml7700_api = {
 	.sample_fetch = veml7700_sample_fetch,
 	.channel_get = veml7700_channel_get,
 	.attr_set = veml7700_attr_set,
 	.attr_get = veml7700_attr_get
 };
 
-#define VEML7700_INIT(n)                                                    \
-	static struct veml7700_data veml7700_data_##n;                      \
-									    \
-	static const struct veml7700_config veml7700_config_##n = {         \
-		.bus = I2C_DT_SPEC_INST_GET(n),                             \
-		.psm = DT_INST_PROP(n, psm_mode)                            \
-	};                                                                  \
-                                                                            \
-	PM_DEVICE_DT_INST_DEFINE(n, veml7700_pm_action);                    \
-                                                                            \
-	SENSOR_DEVICE_DT_INST_DEFINE(n,                                     \
-				     veml7700_init,                         \
-				     PM_DEVICE_DT_INST_GET(n),              \
-				     &veml7700_data_##n,                    \
-				     &veml7700_config_##n,                  \
-				     POST_KERNEL,                           \
-				     CONFIG_SENSOR_INIT_PRIORITY,           \
-				     &veml7700_api);
+#define VEML7700_INIT(n)                                                                           \
+	static const struct veml7700_config veml7700_config_##n = {                                \
+		.bus = I2C_DT_SPEC_INST_GET(n), .psm = DT_INST_PROP(n, psm_mode)};                 \
```

Comment:
Please move the psm field to a new line

Reply:
that's how zephyr's clang-format definition formats code. I added a comma after the psm field to make clang-format move every line into it's own row.